Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -578,7 +578,7 @@ private void UpdateGazeProvider()
             {
                 // The gaze hit result may be populated from the UpdatePointers call. If it has not, then perform
```

Comment:
This is an interesting one. This second pass was introduced to ensure an app could always query what the head/eyes were targeting regardless of if the MRTK input system would send events over it. This change would be a behavioral breaking change if apps are relying on this info while setting the gaze pointer to always be off.

On the other hand...setting the gaze pointer to be _always_ off could imply that it will never be used for anything. Maybe we can take this change and heavily document the meaning.

Reply:
Building on this comment: we specifically have tests that ensure the assertion I made above is true, which this PR is failing: 
<img width="245" alt="image" src="https://user-images.githubusercontent.com/3580640/143321483-dad7b2f5-7640-4655-a0f9-ac592788ebc7.png">

Reply:
Seems like maybe the gaze target should be something lazy evaluated, especially if the behavior is set to AlwaysOff.